pkg/storing/events/events.go

```diff
@@ -174,9 +191,51 @@ func (s *Storage) FindAllStatehashesOnCommonHeight(nodes []string) ([]entities.N
 		nodesHeights = append(nodesHeights, entities.NodeStatement{Node: node, Height: h, Status: entities.OK})
 	}
 
+	return nodesHeights, nodesList, minHeight
+}
+
+func (s *Storage) FindMinCommonSpecificHeight(nodesList map[string]bool, minHeight int) ([]entities.NodeStatement, map[string]bool) {
+	// looking for the min common height and the max height
+	var nodesHeights []entities.NodeStatement
+	for node := range nodesList {
+		found, err := s.FoundStatementAtHeight(node, minHeight)
+		if err != nil || !found {
+			nodesList[node] = false // this node is unreachable
+			nodesHeights = append(nodesHeights, entities.NodeStatement{Node: node, Status: entities.Unreachable})
+			continue
+		}
+		nodesHeights = append(nodesHeights, entities.NodeStatement{Node: node, Height: minHeight, Status: entities.OK})
+	}
+
+	return nodesHeights, nodesList
+}
+
+func (s *Storage) FindAllStatehashesOnCommonHeight(nodes []string) ([]entities.NodeStatement, error) {
+	var heightDifference = depthCommonHeightSearch
```

Comment:
`heightDifference` variable is redundant.

Reply:
It's not in this case. `heightDifference` just happened to be the same as `depthCommonHeightSearch` (10), so I decided to make one constant